### PR TITLE
[WIP] Added JSON serializer support for ActiveSupport::Cache

### PR DESCRIPTION
**Pull request by kogulko**

### CODE REVIEW

File: activesupport/lib/active_support/cache.rb [hunk 1]
```diff
@@ -843,49 +842,60 @@ def [](version)
...
+        private
+          def loaded(payload)
+            Marshal.load(payload)
+          rescue TypeError
+            JSON.decode(payload, symbolize_names: true)
+          end
+
+          def serialize(payload)
+            Marshal.dump(payload)
+          end
```
Comment (fresh-eggs) [inline activesupport/lib/active_support/cache.rb:881]: Is there a reason why we can't name these `serialize` and `deserialize` ?

Comment (kogulko) [reply activesupport/lib/active_support/cache.rb:881]: Sound's good!

File: activesupport/lib/active_support/cache.rb [hunk 2]
```diff
@@ -895,6 +905,25 @@ def dump_compressed(entry, threshold)
...
+            super
+          end
+
+          def loaded(payload)
+            return super if json_serializer?(payload)
+
+            JSON.decode(payload, symbolize_names: true)
+          rescue ::JSON::ParserError
+            Marshal.load(payload)
+          end
```
Comment (fresh-eggs) [inline activesupport/lib/active_support/cache.rb:922]: > Is there a reason why we can't name these `serialize` and `deserialize` ?

Same idea with this and the rest of the coder, keeps things easier to follow.

File: activesupport/lib/active_support/cache.rb (hunk 1 above)
Comment (fresh-eggs) [reply activesupport/lib/active_support/cache.rb:881]: Thinking about this more, what situation are we trying to account for by falling back to JSON given a `TypeError` in the default `loaded` ?

We're overwriting it in the`Rails70Coder` right ?

**Pull request closed without merging**